sched/definitions.go

```diff
@@ -17,3 +17,33 @@ type Task struct {
 	Command    []string // Argv to run
 	SnapshotId string
 }
+
+type JobState int
+
+const (
+	// An unambiguous 0-value.
+	UNKNOWN JobState = iota
```

Comment:
I imagine that I'd send the job, get back its ID (knowing it made it onto the queue), and then I'll have this ID. I'll want to sit there and poll its status, seeing it progress. It would be weird if I got the ID back, then as soon as I poll it's invisible.

An alternative would be to have the initial submission not return at all until the job has been pulled from the queue into the distributor and been persisted to a saga log. That would make this code unnecessary, but in order to give clients a good experience we'd need to make sure a job doesn't stay in the queue for more than... 200 milliseconds?